components/gitpod-db/src/typeorm/auth-code-repository-db.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { log } from '@gitpod/gitpod-protocol/lib/util/logging';
+import { DateInterval, OAuthAuthCode, OAuthAuthCodeRepository, OAuthClient, OAuthScope, OAuthUser } from "@jmondi/oauth2-server";
+import * as crypto from 'crypto';
+import { inject, injectable } from "inversify";
+import { EntityManager, Repository } from "typeorm";
+import { DBOAuthAuthCodeEntry } from './entity/db-oauth-auth-code';
+import { TypeORM } from './typeorm';
+
+const expiryInFuture = new DateInterval("5m");
+
+@injectable()
+export class AuthCodeRepositoryDB implements OAuthAuthCodeRepository {
+
+    @inject(TypeORM)
+    private readonly typeORM: TypeORM;
+
+    protected async getEntityManager(): Promise<EntityManager> {
+        return (await this.typeORM.getConnection()).manager;
+    }
+
+    async getOauthAuthCodeRepo(): Promise<Repository<DBOAuthAuthCodeEntry>> {
+        return (await this.getEntityManager()).getRepository<DBOAuthAuthCodeEntry>(DBOAuthAuthCodeEntry);
+    }
+
+    public async getByIdentifier(authCodeCode: string): Promise<OAuthAuthCode> {
+        log.info(`getByIdentifier ${authCodeCode}`);
```

Comment:
debugging leftover? I reckon this would get pretty loud in prod :)

Reply:
That assumes lots of people will be accessing it initially ;-)

Agreed there is a lot of debug logging - I'll take a pass at removing some of them.